trucks.js

```diff
@@ -1,256 +1,290 @@
   {
     name: 'Chewaya Authentic Moroccan Sanwiches',
     type: 'Moroccan',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Chick\'n Fix',
     type: 'American/Fillipino',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday']
+    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday',],
   },
   {
     name: 'Chopstix',
     type: 'Asian Fusion',
-    schedule: ['Tuesday', 'Wednesday', 'Thursday']
+    schedule: ['Tuesday', 'Wednesday', 'Thursday',],
   },
   {
     name: 'Contigo',
     type: 'Mexican',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Convoy Coffee',
     type: 'Coffee',
-    schedule: ['Wednesday', 'Saturday', 'Sunday']
+    schedule: ['Wednesday', 'Saturday', 'Sunday',],
   },
   {
     name: 'Crisp Creperie',
     type: 'French',
-    schedule: ['Tuesday']
+    schedule: ['Tuesday',],
   },
   {
     name: 'Curb Jumper Street Eats',
     type: 'Sandwiches/Burgers',
-    schedule: ['Monday', 'Wednesday']
+    schedule: ['Monday', 'Wednesday',],
   },
   {
     name: 'Curbside',
     type: 'Vietnamese',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Danielle\'s Crepes',
     type: 'French',
-    schedule: ['Tuesday', 'Friday']
+    schedule: ['Tuesday', 'Friday',],
   },
   {
     name: 'Dante\'s Inferno Dogs',
     type: 'Hot Dogs',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Delicatessen Montanti',
     type: 'Sandwiches',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']
+    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday',],
   },
   {
     name: 'Diablo Food Truckz',
     type: 'Asian',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Djung on Wheels',
     type: 'Thai',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Friday']
+    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Friday',],
   },
   {
     name: 'Dog Japon',
     type: 'Hot Dogs',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+    schedule: ['Monday', 'Tuesday', 'Wednesday',
+    'Thursday', 'Friday', 'Saturday', 'Sunday',],
   },
   {
     name: 'Dogfather Catering',
     type: 'Hot dogs, sausages, wood fired pizza',
-    schedule: []
+    schedule: [],
   },
   {
     name: 'Don Lucho\'s',
     type: 'Peruvian Sandwiches',
-    schedule: []
-  }
+    schedule: [],
+  },
 ];
 
-/* 
+/*
  * return an object from the module with a single method on it: filterByDay
- * that method should take a single parameter that represents the day to filter on
- * use the built-in filter() method to return all trucks that have the day in their
+ * that method should take a single parameter
+ * that represents the day to filter on
+ * use the built-in filter() method to return
+ * all trucks that have the day in their
  * schedule
- */
\ No newline at end of file
+ */
+
+module.exports = {
+
+  filterByDay: function (day) {
+    var testArray;
+    var trucksToday = [];
+
+    function checkTruck(testArray) {
```

Comment:
Avoid declaring functions inside other functions like this. Additionally, avoid the use of the same variable name used for different purposes within the same part of code - in this case `testArray` as it makes the code more difficult to follow.